flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/PushingAsyncDataInput.java

```diff
@@ -57,5 +57,7 @@
 		void emitStreamStatus(StreamStatus streamStatus) throws Exception;
 
 		void emitLatencyMarker(LatencyMarker latencyMarker) throws Exception;
+
+		default void endOutput() throws Exception {}
```

Comment:
Why is this exposed to the things implementing `PushingAsyncDataInput`? I guess to make implementation easier?

I mean, there should be no reason for `StreamTaskNetworkInput` or `StreamTaskSourceInput` to call `output.endOutput()`, as the runtime can do it for them and it can only lead to bugs if the method is called prematurely.

Maybe there is an easy way to hide it from the `emitNext()`? Like splitting into two separate interfaces?

Reply:
Make sense, will hide it from there.